### ISSUE TITLE
the output of FM is always 0

### ISSUE THREAD

**Issue by livmortis** [alibaba/EasyRec#380]
The FM code in /layers/fm.py is:
`    .

      fm_feas = tf.concat(fm_fea, axis=1)

      fm_feas = tf.expand_dims(fm_feas, axis=1)

      sum_square = tf.square(tf.reduce_sum(fm_feas, 1))

      square_sum = tf.reduce_sum(tf.square(fm_feas), 1)

      y_v = 0.5 * tf.subtract(sum_square, square_sum)`

The shape of 'fm_feas' is [b, 1, embedding_dim * fields_num ].  Operation 'reduce_sum' is sum in 1 axis, so it's Ineffective . So that sum_square  always equals square_sum and y_v is always zero.
Furtherly, the shape of y_v is [b,  embedding_dim * fields_num ] , it's output of FM tower  in DeepFM model. It will concat with wide tower and deep tower (if 'finall_dnn' exist),  the shape of these two tower is 1 and a fixed value, but shape of FM is as long as fields number, which could be several thousands. That consumes a lot of computing and storage resources.

**Comment by yangxudong** [comment]
谢谢反馈，问题已修复
